src/migrate/index.js

```diff
@@ -121,21 +139,72 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.boolean('is_locked');
+    });
+  }
+
+  _getLockTableName(tableName) {
+    return tableName + '_lock';
+  }
+
+  _isLocked() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .select('*')
+      .then(function(data) {
+        if (data[0].is_locked) {
+          return true;
+        }
+        return false;
+      });
+  }
+
+  _lockMigrations() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .update({ is_locked: true });
+  }
+
+  _unlockMigrations() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .update({ is_locked: false });
+  }
+
   // Run a batch of current migrations, in sequence.
   _runBatch(migrations, direction) {
-    return Promise.all(_.map(migrations, this._validateMigrationStructure, this))
+    return this._isLocked()
+      .then(isLocked => {
+        if (isLocked) {
+          throw new Error('migrations failed: migration in progress');
```

Comment:
Shouldn't this be LockingError thrown to prevent releasing the lock? This case should be catched by the test cases.